.github/workflows/ci-tests.yml

```diff
@@ -32,40 +32,49 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os:
-          - ubuntu-22.04
-          - macos-12
-          - windows-2022
+        os: ${{ fromJson(github.repository_owner == 'subspace' && github.repository == 'subspace/subspace-cli' && '[["self-hosted", "ubuntu-22.04-x86-64", "Linux"], ["self-hosted", "macos-12-arm64", "macOS"], ["self-hosted", "windows-server-2022-x86-64", "Windows"]]' || '["ubuntu-22.04", "macos-12", "windows-2022"]') }}
         run-all:
           - ${{ inputs.test-macos-and-windows == true || github.ref == 'refs/heads/main' }}
         exclude: # exclude macos-12 and windows-2022 when the condition is false
           - run-all: false
             os: macos-12
           - run-all: false
             os: windows-2022
+          - run-all: false
+            os: macOS
+          - run-all: false
+            os: Windows
```

Comment:
I don't think this is working, there are no such `os` values in the matrix above. You also need to find the reason these excludes are present in the first place, maybe it was because they were slow, which is not relevant anymore.

Reply:
Those are targeted from the self-hosted runner labels, which can be cross-checked with and example 
`if: runner.os == 'Windows'`

Reply:
You may be confusing `matrix.os` vs `runner.os`, which are not the same thing. Here we deal with `matrix.os`.

Reply:
changed this to
```
      matrix:
        os: ${{ fromJson(github.repository_owner == 'subspace' && '[["self-hosted", "ubuntu-20.04-x86-64", "Linux"], ["self-hosted", "macos-12-arm64", "macOS"], ["self-hosted", "windows-server-2022-x86-64", "Windows"]]' || '["ubuntu-20.04", "macos-12", "windows-2022"]') }}
        run-all:
          - ${{ inputs.test-macos-and-windows == true || github.ref == 'refs/heads/main' }}
        exclude: # exclude macos-12 and windows-2022 when the condition is false
          - run-all: false
            os: macos-12
          - run-all: false
            os: windows-2022
```